android/src/main/java/cz/firest/calendar/CapacitorCalendar.java

```diff
@@ -127,7 +127,8 @@ protected void createCalendarEvent(PluginCall call) {
         try {
             Long startTime = data.has("startDate") ? data.getLong("startDate") : new Date().getTime();
             Long endTime = data.has("endDate") ? data.getLong("endDate") : new Date().getTime();
-
+            final boolean allDayEventConfig = data.getBoolean("allDay", true);
+            final boolean allDayEvent = allDayEventConfig && isAllDayEvent(new Date(startTime), new Date(endTime));
```

Comment:
Now I'm a little confused with the meaning of changes. So, you want to set that event is NOT allDay, even if it's allDay by checking the event range? Or you want to set the event as allDay even it's not allDay by checking the event range?

Reply:
Indeed the goal is to not create an all day event when `isAllDayEvent` returns true.

Maybe the naming of the property in the config is not appropriate.

Reply:
Ok. As I said, I think it would be best to just use _isAllDayEvent_ in case that _allDay_ option is not specified. This allows you to use the required functionality but also keeps the original functionality (required by our projects).